core/src/main/scala/kafka/api/FetchResponse.scala

```diff
@@ -48,6 +49,14 @@ object FetchResponsePartitionData {
 
 case class FetchResponsePartitionData(error: Short = Errors.NONE.code, hw: Long = -1L, messages: MessageSet) {
   val sizeInBytes = FetchResponsePartitionData.headerSize + messages.sizeInBytes
+
+  def toMessageFormat(toMagicValue: Byte): FetchResponsePartitionData = {
+    if (messages.hasMagicValue(toMagicValue))
```

Comment:
This means that we are paying the overhead of checking hasMagicValue even after the consumer is upgraded to support both message format v0 and v1.